account_financial_report_qweb/report/trial_balance.py

```diff
@@ -247,9 +251,16 @@ def _inject_account_values(self, account_ids):
 WHERE
     acc.id in %s
         """
-        if self.hide_account_balance_at_0:
-            query_inject_account += """ AND
-    final_balance IS NOT NULL AND final_balance != 0"""
+        if self.hide_account_at_0:
+            query_inject_account += """ AND (
+    (initial_balance IS NOT NULL AND initial_balance != 0) OR
```

Comment:
I wouldn't test about initial balance and final_balance, as you can have an initial balance but no activity on the period (so you will also have final balance), but it's not too much relevant for the trial balance. Don't you think?